core/domain/user_services_test.py

```diff
@@ -307,6 +338,49 @@ def test_get_user_id_from_username(self):
         self.assertIsNone(
             user_services.get_user_id_from_username('fakeUsername'))
 
+    def test_get_user_settings_from_username_returns_user_settings(self):
+        auth_id = 'someUser'
+        username = 'username'
+        user_email = 'user@example.com'
+
+        user_id = user_services.create_new_user(auth_id, user_email).user_id
+        user_services.set_username(user_id, username)
+        user_settings_model = user_models.UserSettingsModel.get_by_id(user_id)
+        user_settings = user_services.get_user_settings_from_username(username)
+
+        self.assertEqual(
+            user_settings_model.id, user_settings.user_id)
+        self.assertEqual(
+            user_settings_model.email, user_settings.email)
+
+    def test_get_user_settings_from_username_for_no_username_is_none(self):
+        auth_id = 'someUser'
+        username = 'username'
+        user_email = 'user@example.com'
+
+        user_id = user_services.create_new_user(auth_id, user_email).user_id
+        user_services.set_username(user_id, username)
+
+        self.assertIsNone(
+            user_services.get_user_settings_from_username('fakeUsername'))
+
+    def test_get_user_settings_from_email_returns_user_settings(self):
+        auth_id = 'someUser'
+        user_email = 'user@example.com'
+
+        user_id = user_services.create_new_user(auth_id, user_email).user_id
+        user_settings_model = user_models.UserSettingsModel.get_by_id(user_id)
+        user_settings = user_services.get_user_settings_from_email(user_email)
+
+        self.assertEqual(
+            user_settings_model.id, user_settings.user_id)
+        self.assertEqual(
+            user_settings_model.email, user_settings.email)
```

Comment:
Ditto re hardcoding expected values (or using user_email from above).

Reply:
I have the same problem here as the issue above. While I can hardcode the expected value for emails, I can't do the same for user ids.

Reply:
Ditto as above.